### PR TITLE
Refactor LLMChatBlock to extend BaseBlock and enhance validation logic

### PR DESCRIPTION
**Pull request by shivchander**
- Change LLMChatBlock to inherit from BaseBlock instead of Block for improved functionality.
- Implement custom validation logic for message formats, ensuring messages are lists and contain required fields.
- Update input validation method to provide detailed error messages for better debugging.
- Add comprehensive tests for custom validation scenarios, including various invalid message formats and edge cases.

This refactor streamlines the validation process and enhances the robustness of the LLMChatBlock, ensuring it meets the expected input structure.

### CODE REVIEW

File: src/sdg_hub/blocks/llm/llm_chat_block.py [hunk 1]
```diff
@@ -358,62 +381,69 @@ def get_model_info(self) -> Dict[str, Any]:
...
+                    raise BlockValidationError(
+                        f"Message {msg_idx} in row {idx} missing required 'content' field",
+                        details=f"Block: {self.block_name}, Row: {idx}, Message: {msg_idx}, Available fields: {list(message.keys())}",
+                    )
+
+            return True  # Return something for map
+
+        # Use map to validate all samples
+        # Add index to each sample for better error reporting
+        indexed_samples = [(i, sample) for i, sample in enumerate(dataset)]
```
Comment (eshwarprasadS) [inline src/sdg_hub/blocks/llm/llm_chat_block.py:445]: can we simplify this map call `map(validate_sample, indexed_samples)` and accordingly change the signature for the `validate_sample` method? right now we package the index and sample into a parameter, pass it to the method, and split it again into idx, sample, which could simply be two parameters to pass in the first place, for readability.

**Review by eshwarprasadS — APPROVED** [review]
Just a minor comment, everything else looks good!
